### PR TITLE
Nuke prolematic file on Azure

### LINKED ISSUE #112: [BUG] The `yum` RPM is wonky on Azure
**Describe the bug**

When the `OSpackages.sh` script attempts to reinstall the `yum` RPM, the unpacking of the RPM fails with an error like:

~~~bash
  Installing       : yum-4.7.0-19.el8.noarch                            223/231Error unpacking rpm package yum-4.7.0-19.el8.noarch

  Installing       : python3-syspurpose-1.28.40-1.el8_9.x86_64          224/231
error: unpacking of archive failed on file /etc/yum/pluginconf.d;65df690f: cpio: File from package already exists as a directory in system
error: yum-4.7.0-19.el8.noarch: install failed
~~~

**Severity**

Completely Broken (No work-around evident)

**Expected behavior**

`OSpackages.sh` script executes normally as it did prior to January 2024 and as it does in AWS

**Fix Suggestions**

Add logic to the `OSpackages.sh` script to conditionally delete the `/etc/yum/pluginconf.d` directory on Azure-based systems. Inserting logic similar to:

~~~bash
if  [[ $( dmidecode --string chassis-asset-tag ) == "7783-7084-3265-9085-8269-3286-77" ]] &&
    [[ -d /etc/yum/pluginconf.d ]]
then
    err_exit "Deleting /etc/yum/pluginconf.d on an Azure build-host..." NONE
    rm -rf /etc/yum/pluginconf.d || err_exit "Failed deleting /etc/yum/pluginconf.d. Aborting... " 1
    err_exit "Successfully deleted /etc/yum/pluginconf.d" NONE
fi
~~~

Before the [execution of](https://github.com/plus3it/amigen8/blob/master/OSpackages.sh#L267-L271) the `yum reinstall` command appears to work past the issue when the `OSpackages.sh` script  is manually-executed.

Notifying @nortiz34 for awareness

### PR DESCRIPTION
**Pull request by ferricoxide**
Closes #112 (as originally opened on behalf of @nortiz34)

### CODE REVIEW

**Comment by ferricoxide** [comment]
Pending validation-testing in Azure...

File: OSpackages.sh [hunk 1]
```diff
@@ -375,6 +378,34 @@ function FetchCustomRepos {
 
 }
 
+# Prevent Azure-specific failure
+function AzureYumPluginDirCollision {
+  local AZURE_ASSET_TAG
+  local YUM_PROBLEM_DIR
+
+  # The asset-tag value for dmidecode
+  AZURE_ASSET_TAG="7783-7084-3265-9085-8269-3286-77"
```
Comment (lorengordon) [inline OSpackages.sh:387]: Do we have any links to Azure docs as a source for this value? Thinking ahead to when we look at this in a couple years and wonder how we got the value.

Comment (ferricoxide) [reply OSpackages.sh:387]: Other than StackExchange? Lemme go look.

Comment (ferricoxide) [reply OSpackages.sh:387]: I'll add this URL:

~~~
https://learn.microsoft.com/en-us/azure/automation/troubleshoot/update-agent-issues-linux#dmidecode-check
~~~

Into the script.

**Pull request merged**